### PR TITLE
Add Meilisearch example

### PR DESCRIPTION
**Pull request by Strift**
Hello Next.js maintainers 👋 

### What?

- Adding an example folder for Meilisearch starter with Next.js

### Why?

- Ease the integration of Next.js developers using Meilisearch
- Support the launch of the Meilisearch Vercel integration

### How?

- Adding a `with-meilisearch` folder in `examples/`

I hope this PR matches your expectations regarding new examples. Let me know if I need to update anything. 🙏 

Thanks in advance.

### CODE REVIEW

**Comment by Strift** [comment]
Hi, 

I have updated the Deploy button to remove the integration ID from it. 

I will make a follow-up PR to update it when the Meilisearch Vercel integration is officially released.

File: examples/with-meilisearch/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+Deploy this project on Vercel:
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/clone?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-meilisearch&project-name=with-meilisearch&repository-name=with-meilisearch)
+
+## How to use
+
+1. Create a [Meilisearch Cloud project](https://cloud.meilisearch.com)
+1. Run `npx create-next-app --example with-meilisearch my-app` to create a Next.js app using the Meilisearch starter kit
+1. Navigate to your application directory, eg. `cd my-app`
```
Comment (leerob) [inline examples/with-meilisearch/README.md:15]: Please follow the structure of the other example READMEs: https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md

File: examples/with-meilisearch/tailwind.config.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
```
Comment (leerob) [inline examples/with-meilisearch/tailwind.config.js:2]: Should we use `tailwind.config.ts` here?

File: examples/with-meilisearch/README.md (hunk 1 above)
Comment (Strift) [reply examples/with-meilisearch/README.md]: I've updated it to match the new guidelines for the "How to use" section.

I still had to include a notice about the user needing an account/credentials.

l hope that's ok? let me know if it isn't :)

File: examples/with-meilisearch/tailwind.config.js (hunk 2 above)
Comment (Strift) [reply examples/with-meilisearch/tailwind.config.js:2]: Moved to TS :)

**Comment by Strift** [comment]
Hi @leerob, thank you for your review!

I have made some changes to the project to address your feedback:
- Update README
- Migrate TailwindCSS configuration file to Typescript

I hope it's good now :)

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: cdf6b3155d43ed9632ed7989e52f2f978e79cd65

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: cdf6b3155d43ed9632ed7989e52f2f978e79cd65

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by leerob — DISMISSED** [review]

**Comment by leerob** [comment]
```
/root/actions-runner/_work/next.js/next.js/examples/with-meilisearch/env.d.ts
  1:11  warning  'NodeJS' is defined but never used      @typescript-eslint/no-unused-vars
  2:13  warning  'ProcessEnv' is defined but never used  @typescript-eslint/no-unused-vars

✖ 2 problems (0 errors, 2 warnings)
```

**Review by Strift — COMMENTED** [review]
Apologies for the oversight.

I've added the missing `declare` statement. This fixed the linting issue.

**Review by leerob — DISMISSED** [review]

**Review by leerob — APPROVED** [review]

**Review by leerob — APPROVED** [review]

File: examples/with-meilisearch/package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "scripts": {
+    "dev": "next dev",
```
Comment (leerob) [inline examples/with-meilisearch/package.json:3]:
```suggestion
    "dev": "next dev --turbo",
```

**Comment by leerob** [comment]
Could you run Prettier again please?

**Review by leerob — APPROVED** [review]

File: examples/with-meilisearch/package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+  "scripts": {
```
Comment (leerob) [inline examples/with-meilisearch/package.json:2]:
```suggestion
  "private": true,
  "scripts": {
```

**Comment by Strift** [comment]
Hey, I ran prettier and updated all dependencies as agreed :)

**Pull request merged**